library/ssl_tls.c

```diff
@@ -4219,12 +4219,12 @@ void mbedtls_ssl_handshake_free(mbedtls_ssl_context *ssl)
     mbedtls_ssl_buffering_free(ssl);
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
-#if defined(MBEDTLS_PK_CAN_ECDH) && \
+#if defined(PSA_WANT_ALG_ECDH) && \
     (defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3))
     if (handshake->ecdh_psa_privkey_is_external == 0) {
```

Comment:
I'm just wondering about `ecdh_psa_privkey_is_external` guards here and in `ssl_misc.h`: `(MBEDTLS_ECDH_C || MBEDTLS_ECDSA_C) && `(MBEDTLS_USE_PSA_CRYPTO || MBEDTLS_SSL_PROTO_TLS1_3)` vs `(PSA_WANT_ALG_ECDH) && (MBEDTLS_USE_PSA_CRYPTO || MBEDTLS_SSL_PROTO_TLS1_3)`:
This would be a problem if we would have `MBEDTLS_USE_PSA_CRYPTO` on, but not `MBEDTLS_ECDH_C` or `MBEDTLS_ECDSA_C`. 
You can test it by adding `scripts.config.py set MBEDTLS_USE_PSA_CRYPTO` after `# start with default + driver support` in `component_test_psa_crypto_config_accel_ecc`.
If that's resolved in future PRs then I'm sorry, as I haven't looked into all of them yet! :)

Reply:
You're right, this is an error. Thanks for noticing! I will fix it

Reply:
>  If that's resolved in future PRs then I'm sorry, as I haven't looked into all of them yet! :)

Yes, there is another issue (#7272) which closes the series about accelerated EC drivers, but IMO each PR should be "correct" on its own and this is not ;)